app/apollo/resolvers/resource.js

```diff
@@ -29,6 +29,33 @@ const conf = require('../../conf.js').conf;
 const S3ClientClass = require('../../s3/s3Client');
 const url = require('url');
 
+// Filters out the namespaces you dont have access to. has to get all the resources first.
+const filterNamespaces = async (data, me, orgId, queryName, context) => {
+  const res = data.resources.resources;
+  const deleteArray = [];
+  const namespaces = res.map(d => d.searchableData.namespace).filter((v, i, a) => a.indexOf(v) === i).filter(x => x);
+
+  //need to adjust validateAuth to retun false and not throw error
+  namespaces.forEach(n => {
+    async () => {
+      const valid = await validAuth(me, orgId, ACTIONS.READ, TYPES.RESOURCE, queryName, context, [n]);
+      if (!valid) deleteArray.push(n);
+    };
+  });
```

Comment:
js foreach's dont run `await` on the called function. So `deleteArray` will always be `[]` at the moment the foreach finishes.

`await Promise.all(namespaces.map(async()=>{ ... }))` would work for the most part.

Although we might want to put concurrency limits on it, in case we have 1000 namespaces. Other parts of the code use the `p-limit` library for that.

Reply:
🥇  for the promise suggestion.  I really didnt want to put in that foreach.